### DISCUSSION TITLE
(Feature Request) Implement a "Reboot" command

### DISCUSSION THREAD

**Discussion by Stoicmaverick** [arendst/Tasmota#15193] [Ideas]
I love the project, and I have a ton of Tasmota flashed devices interfacing with HA around the house, but exactly ONE thing sets my teeth on edge: If (for a variety of reasons) I find myself needing to reboot a device via the console CLI, I spend about five minutes quadruple checking the command before I hit 'Enter'. Many of the devices are located in very inaccessible locations, like behind light switches, or in the attic, and while 'Restart 1' is the command that I want to run, my tired brain typing 'Restart 2' or, god forbid 'Reset 1' to every device on the network via TasmotaAdmin, would be a messy weekend to recover from.

I am inevitably doing this kind of maintenance at 11:45pm so as not to disrupt people using the devices, so this is something I find myself thinking about. It would be nice if there was a simple "Reboot" command in the CLI that would take the tension out of such a simple thing.

I am not a coder, so this is far beyond my skill to PR, but is there a place to enter a feature request, or has this ever come up internally? It seems like it would be simple alias of 'Reboot="Restart 1"', and I confidently base that on almost nothing.

Thanks for the project.

**Comment by barbudor** [comment]
One of the good reasons to have a parameter after restart is also to avoid unwanted restart.
Just like `Upgrade` requires to have `1` after the command, it sounds reasonnable to have a failsafe on `Restart` requiring an additional parameter to be specified.

There are still different options that could allow you to make it easier for you :
1. Create an event with a rule
  `ON event#reboot DO restart 1 ENDON`
So you just have to type `event reboot` in the console.

2. If you instist on typing only `Reboot`, [Scripting-Language](https://tasmota.github.io/docs/Scripting-Language) on ESP8266/ESP32 or [Berry](https://tasmota.github.io/docs/Berry/) on ESP32 will allow you to create you own commands.
 - Scripting : search for USE_SCRIPT_SUB_COMMAND 
 - Berry : https://tasmota.github.io/docs/Berry/#functions-to-create-custom-tasmota-command

3. Why do you want to do it manually in the console when it is so easy to schedule or control it it from you back-end/home-automation through either MQTT or event HTTP request ?
Even if you don't have an home-automation system, you can easily create a short-cut/script or whatever to send the command to the right device in a single click.

**Reply by sfromis** [reply]
For ESP8266-based devices, one other way of adding a command which is not quite what you ask for (only works via MQTT) is:

If you use a build where the optional [Subscribe](https://tasmota.github.io/docs/Subscribe-&-Unsubscribe/) is added, you can also set up a boot rule to subscribe to a new command name `Subscribe reboot,cmnd/your.tasmota.topic/reboot` and a rule like `on Event#reboot do Restart 1 endon`. This allows a command via MQTT (like from TasmoAdmin), but not directly in the console, and only works if the command has an argument, like publishing to `cmnd/your.tasmota.topic/reboot` with the payload `1` or whatever.

**Comment by Jason2866** [comment]
I see no benefit by doing this. There is still the chance to mess up the commands and do a `reset 1` by accident or other commands you can use by accident which do unwanted.
Safety belt solutions are already sugessted.

**Reply by Stoicmaverick** [reply]
Well you're right that it's impossible to prevent a reckless degree of carelessness, but, for instance, most linux distros that I am familiar with alias "reboot" with "shutdown -r" for this reason. It's a pain to accidentally power down a remote system via ssh that you don't have easy physical access to.

It's also one reason that it's inadvisable to run as 'root' for routine tasks. 

I'm curious what you mean by "safety belt solutions" because to me, this would be a safety belt insomuch as it's a small degree of compartmentalization for the user. "reboot 1" would be fine, but without looking up the command list on the website, you would know that reboot=safe, but restart=watch out.

**Comment by jeroenst** [comment]
I now have a remote tasmota device with malfunctioning berry script. A reboot cmnd would do the trick for now but because this command is not implemented the people at the remote location will get a warm house tomorrow morning when it will be 30 degrees outside. Just a reboot via mqtt would prevent this..

**Reply by s-hadinger** [reply]
To reboot just do `Restart 1` via MQTT

**Reply by sfromis** [reply]
If you have a need to "suspend" Berry executing after the `Restart 1` (which is the regular way of doing a reboot), you can also use 'SetOption153 1` to disable Berry `autoexec.be` from running.

In addition, the command `BrRestart` exists to restart the Berry VM without rebooting Tasmota. Obviously also available via MQTT.

**Reply by jeroenst** [reply]
Thank you!
